spring-cloud-gcp-logging/src/main/java/org/springframework/cloud/gcp/logging/StackdriverJsonLayout.java

```diff
@@ -0,0 +1,83 @@
+/*
+ *  Copyright 2018 original author or authors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.springframework.cloud.gcp.logging;
+
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+import ch.qos.logback.classic.spi.ILoggingEvent;
+import ch.qos.logback.classic.spi.IThrowableProxy;
+import com.google.cloud.logging.TraceLoggingEnhancer;
+import com.google.gson.Gson;
+
+/**
+ * @author Andreas Berger
+ */
+public class StackdriverJsonLayout extends ch.qos.logback.contrib.json.classic.JsonLayout {
+
+	public static final String SEVERITY_ATTR_NAME = "severity";
+
+	public static final String ATTR_NAME_TIMESTAMP_SECONDS = "timestampSeconds";
+
+	public static final String ATTR_NAME_TIMESTAMP_NANOS = "timestampNanos";
+
+	private final String projectId;
+
+	public StackdriverJsonLayout() {
+		setAppendLineSeparator(true);
+		setIncludeMessage(false);
+		setIncludeException(false);
+		setIncludeFormattedMessage(false);
+		Gson formatter = new Gson();
+		setJsonFormatter(formatter::toJson);
+		this.projectId = System.getenv("GOOGLE_CLOUD_PROJECT");
```

Comment:
I think this has to be configurable: https://docs.spring.io/spring-boot/docs/2.0.0.RC2/reference/htmlsingle/#_environment_properties

As we as all other properties, too. IMHO...

Reply:
Not sure what you mean about `no configuration scanning`, but what you have now is really just hard code and in Spring Boot no body is going to always set ENV var. You may have it as a fallback though...